### PR TITLE
Refactor visualizer cloning around per-group ClonePlan

### PR DESCRIPTION
**Pull request by ooctipus**
# Description

Routes the visualizer-side cloning data through a per-group `ClonePlan` map and has the PhysX scene data provider pull a Newton model from those plans directly — replacing the clone-time `visualizer_clone_fn` callback and the `VisualizerPrebuiltArtifacts` payload introduced in #5398.

## Pipeline

```
InteractiveScene.clone_environments
    plans = clone_from_template(...)        # dict[str, ClonePlan]
    sim.set_clone_plans(plans)              # canonical owner
            ↓
PhysxSceneDataProvider.__init__
    plans = sim.get_clone_plans()
    sources, destinations, mask = aggregate(plans)
    positions = read_env_xforms(stage)
    model, state = newton_visualizer_prebuild(...)
```

Single source of truth, single direction. No clone-time callback, no requirements push from sensors, no late-resolve refresh loop.

## Notable changes

- **Add** `ClonePlan` (`dest_template`, `prototype_paths`, `clone_mask`); `SimulationContext.{get,set}_clone_plans`; `InteractiveScene.clone_plans` forwarder.
- **Remove** `TemplateCloneCfg.visualizer_clone_fn`, `cloner.resolve_visualizer_clone_fn`, `VisualizerPrebuiltArtifacts`, `SimulationContext.{get,set,clear}_scene_data_visualizer_prebuilt_artifact`, `Camera._register_renderer_scene_data_requirements`, `create_newton_visualizer_prebuild_clone_fn`.
- **Collapse** three calls to `_refresh_visualizer_clone_fn_from_requirements` into one `_aggregate_scene_data_requirements` after entities are constructed.
- **Provider** derives flat `(sources, destinations, mask)` from per-group plans, recovering each source path as `dest_template.format(<first env using this prototype>)` and reading per-env positions off the env-template's `xformOp:translate`.

## Type of change

- Bug fix
- Breaking change *(removes the public symbols listed above. They were introduced in #5398 four days ago; the changelog fragments are filed as `.minor.rst` since the surface had no time to acquire dependents.)*

## Checklist

- [x] I have read and understood the contribution guidelines
- [x] I have run the `pre-commit` checks with `./isaaclab.sh --format`
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added changelog fragments under `source/isaaclab/changelog.d/`, `source/isaaclab_newton/changelog.d/` (both `.minor.rst`), and `source/isaaclab_physx/changelog.d/` (`.skip` — private/internal changes only)
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

File: source/isaaclab_physx/isaaclab_physx/scene_data_providers/physx_scene_data_provider.py [hunk 1]
```diff
@@ -164,26 +166,96 @@ def _model_body_paths(self, model) -> list[str]:
...
+            num_envs = plan_list[0].clone_mask.size(1)
+            if any(p.clone_mask.size(1) != num_envs for p in plan_list):
+                raise RuntimeError(
+                    f"[PhysxSceneDataProvider] Clone plans disagree on num_envs: "
+                    f"{[p.clone_mask.size(1) for p in plan_list]}"
+                )
+            sources = [
+                p.dest_template.format(int(p.clone_mask[i].nonzero(as_tuple=False)[0].item()))
+                for p in plan_list
+                for i in range(p.clone_mask.size(0))
```
[the remark this answers is not in the document]
Comment (ooctipus) [reply source/isaaclab_physx/isaaclab_physx/scene_data_providers/physx_scene_data_provider.py]: Fixed in 4abc82ac472. The comprehension now skips all-False rows; new regression test `test_build_from_clone_plans_skips_unused_prototype_rows` exercises a 3-prototype/2-env sequential layout where proto_2 is unassigned.

File: source/isaaclab/isaaclab/cloner/clone_plan.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    """Destination path template for this group, e.g. ``"/World/envs/env_{}/Object"``."""
+
+    prototype_paths: list[str]
+    """Prototype prim paths in this group, e.g.
+    ``["/World/template/Object/proto_asset_0", "/World/template/Object/proto_asset_1"]``."""
+
+    clone_mask: torch.Tensor
+    """Boolean tensor of shape ``[num_prototypes_in_group, num_envs]``;
+    ``clone_mask[i, j]`` is ``True`` iff env ``j`` was populated from
+    :attr:`prototype_paths` ``[i]``. Each column sums to exactly one."""
```
[the remark this answers is not in the document]
Comment (ooctipus) [reply source/isaaclab/isaaclab/cloner/clone_plan.py:39]: Fixed in 4abc82ac472. `prototype_paths` and `clone_mask` are now `field(hash=False, compare=False)` so hash and equality fold in only `dest_template` (the natural identity — it is the dict key in `SimulationContext.get_clone_plans`). New test `test_clone_plan_is_hashable_with_unhashable_fields` asserts `hash(plan)` no longer raises.

File: source/isaaclab_physx/isaaclab_physx/scene_data_providers/physx_scene_data_provider.py [hunk 3]
```diff
@@ -164,26 +166,96 @@ def _model_body_paths(self, model) -> list[str]:
...
+        """
+        positions = torch.zeros((num_envs, 3), dtype=torch.float32, device=self._device)
+        for i in range(num_envs):
+            prim = self._stage.GetPrimAtPath(f"/World/envs/env_{i}")
+            if not prim.IsValid():
+                continue
+            value = prim.GetAttribute("xformOp:translate").Get()
+            if value is not None:
+                positions[i] = torch.tensor([value[0], value[1], value[2]], device=self._device)
+        return positions
```
[the remark this answers is not in the document]
Comment (ooctipus) [reply source/isaaclab_physx/isaaclab_physx/scene_data_providers/physx_scene_data_provider.py]: Fixed in 4abc82ac472. The provider now derives the env path template via `_env_path_template_from_plan(plan_list[0])` (splits `dest_template` on the first `{}` and reattaches the placeholder), then threads it into `_read_env_origins_from_stage`. New test `test_build_from_clone_plans_uses_dest_template_for_env_lookup` exercises a non-default `/Stage/scenes/env_<i>` layout.

**Comment by ooctipus** [comment]
Addressed all four greptile review comments in 4abc82ac472:

| # | Severity | Issue | Fix |
|---|---|---|---|
| 1 | P1 | `IndexError` when a prototype row is all-False (e.g. `num_protos > num_envs` under sequential strategy) | Provider's flatten loop now skips rows where `clone_mask[i].nonzero().numel() == 0`. New regression test `test_build_from_clone_plans_skips_unused_prototype_rows`. |
| 2 | P2 | `@dataclass(frozen=True)` + `torch.Tensor` field → unhashable at runtime | `prototype_paths` and `clone_mask` marked `field(hash=False, compare=False)`; hash/eq fold in only `dest_template` (the natural identity — it is the dict key). New test `test_clone_plan_is_hashable_with_unhashable_fields`. |
| 3 | P2 | Hardcoded `/World/envs/env_{i}` in `_read_env_origins_from_stage` | Added `_env_path_template_from_plan` that derives the env template from `plan.dest_template.split("{}")[0] + "{}"`. Threaded through `_read_env_origins_from_stage(num_envs, env_path_template)`. New test `test_build_from_clone_plans_uses_dest_template_for_env_lookup` exercises a non-default `/Stage/scenes/env_<i>` layout. |
| 4 | P2 | `get_clone_plans`/`set_clone_plans` typed `dict[str, Any]`, defeating static checking | Imported `ClonePlan` under `TYPE_CHECKING` to avoid the cloner→sim circular risk; annotated `_clone_plans`, `get_clone_plans`, `set_clone_plans` as `dict[str, ClonePlan]`. |

Test count: 6/6 (3 original + 3 new regression).

**Review by matthewtrepte — APPROVED** [review]
looks good! thanks for cleaning up this step
